### PR TITLE
Bump dependencies and dev tool versions.

### PR DESCRIPTION
**Pull request by smite**
* ``iqm-client >= 15.2``

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
 # -- General configuration ---------------------------------------------------
 
 # require a recent version of Sphinx
-needs_sphinx = '4.4'
+needs_sphinx = '6.2'
```
Comment (rbrazinskas) [inline docs/conf.py:43]: ``tox -e docs`` breaks on my local setup:

```
    title = caption % part
            ~~~~~~~~^~~~~~
TypeError: not all arguments converted during string formatting
```

I am able to fix it by changing ``extlinks`` dictionary in ``docs/conf.py`` to

```
extlinks = {
    'issue': ('https://github.com/iqm-finland/qiskit-on-iqm/issues/%s', 'issue %s'),
    'mr': ('https://github.com/iqm-finland/qiskit-on-iqm/pull/%s', 'MR %s'),
}
```

**Review by rbrazinskas — COMMENTED** [review]
Small issue with buildin the docs (at least on my local Windows development environment. Otherwise LGTM

File: docs/conf.py (hunk 1 above)
Comment (smite) [reply docs/conf.py:43]: Fixed.

**Review by rbrazinskas — APPROVED** [review]

File: tests/test_bell_measure_example.py [hunk 2]
```diff
@@ -17,12 +17,13 @@
...
 
 import subprocess
+import sys
 
 
 def test_bell_measure_example_call():
     """Test that example script runs and fails at establishing a connection."""
     with subprocess.Popen(
-        ("python", "examples/bell_measure.py", "--cortex_server_url", "https://not.a.real.domain"),
+        (sys.executable, "examples/bell_measure.py", "--cortex_server_url", "https://not.a.real.domain"),
```
Comment (rbrazinskas) [inline tests/test_bell_measure_example.py:26]: @kukushechkin, I would be really curious to understand why did the test break on Windows. Any ideas? :-)

Comment (kukushechkin) [reply tests/test_bell_measure_example.py:26]: I guess the the "parent" python environment (comes from tox I guess?) is different from whatever we call by just calling "python" (comes from venv I guess?). This was it seems to be more consistent, it is not any random python, it is the same python that runs the tests.

**Review by kukushechkin — APPROVED** [review]

**Pull request merged**